graph/src/components/store.rs

```diff
@@ -217,6 +217,12 @@ pub enum EntityCollection {
     /// column `b`; they will be grouped by using `A.a` and `B.b` as the keys
     Window(Vec<EntityWindow>),
 }
+/// The type we use for block numbers. This has to be a signed integer type
+/// since Postgres does not support unsigned integer types. But 2G ought to
+/// be enough for everybody
+pub type BlockNumber = i32;
```

Comment:
I did :smiley:  At 4 blocks/minute, block 2,147,483,647 will be mined in the year 3036. Even for a chain 100x as fast as Ethereum, it would take 10 years to exhaust the `i32` space.

This is also something that was introduced with relational schemas as block ranges are a pair of `int4` in the database, changing that would require a migration of data. Seeing how much runway we have there, there's no need to tackle that now.

Reply:
Fair enough 😁